manifests/resource/geo.pp

```diff
@@ -63,6 +63,10 @@
   Optional[Boolean] $proxy_recursive  = undef
 ) {
 
+  if ! defined(Class['nginx']) {
```

Comment:
If or unless is IMHO very opinionated. I for myself prefer if over unless in most situations. But I'll change to unless if requested.